### ISSUE TITLE
Figure out how to have a browser window

### ISSUE THREAD

**Issue by alivenotions** [mathiasberggren/rush#3]
Since _most_ browsers have windows, we should also have one. This ticket should just be about opening that window

**Comment by mathiasberggren** [comment]
In progress on branch gui-skeleton

**Comment by alivenotions** [comment]
@mathiasberggren This is great work! But I think using a webview does all the work for us 😅 I was wondering if we should look into some more low level crates so we can write the rendering engine ourselves.
Also, can we use this crate and use our own "view" instead of the webview? Because it would be nice to have this for the whole toolbar bit.

**Comment by mathiasberggren** [comment]
The webview is just me playing around trying to check if the behaviour is legit. 

As I understood the cacao crate is mainly a wrapper around AppKit. not sure what limitations that puts on us when building the rendering engine but I think it's a nice place to start from at least :D

**Comment by mathiasberggren** [comment]
Currently following this tutorial for creating windows with Vulkano
https://vulkano.rs/guide/windowing/introduction

**Comment by alivenotions** [comment]
I think I'll also hop on #16 tonight and try things out.

**Comment by mathiasberggren** [comment]
Think I'm approaching being able to render stuff. Not there yet though

**Comment by alivenotions** [comment]
Can you explain a bit what all is doing what on that branch? 😅 
Or maybe I can try to grok it 🤔

**Comment by mathiasberggren** [comment]
Yes for sure! I think we could wait a bit to go through it together until there's something up that works. Right now the only thing working is displaying the screen 😅

**Comment by mathiasberggren** [comment]
I think the tutorial I linked above goes through most of the things if you wanna gronk some with it, it's from there I've taken most ideas
